### PR TITLE
Fix #629

### LINKED ISSUE #629: Defect: Resolving library path variable in the generated script setup.sh
[issue stat img]: https://img.shields.io/issuestats/i/github/sourceryinstitute/OpenCoarrays.svg?style=flat-square

| Avg response time              |
|:------------------------------:|
| ![Issue Stats][issue stat img] |

# Defect/Bug Report #

- OpenCoarrays Version: caf version 2.4.0-6-gebc6f30
- Fortran Compiler: gfortran 8.2.0
- C compiler used for building lib: gcc 8.2.0
- Installation method: `install.sh`
- Output of `uname -a`: Linux v.radia.run 4.18.19-100.fc27.x86_64 #1 SMP Wed Nov 14 22:04:34 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- MPI library being used: MPICH 3.3
- Machine architecture and number of physical cores: MacBookPro running a Linux VM
- Version of CMake: cmake version 3.11.2

## Observed Behavior ##
In my OpenCoarrays setup.sh—which is generated automatically during the build and install process—I see the following stanza:
```
   # Prepend the compiler library paths to the LD_LIBRARY_PATH environment variable:
   if [[ -z "${!LD_LIB_P_VAR}" ]]; then
     export LD_LIBRARY_PATH="/opt/gcc/8.2.0/lib64/:/opt/gcc/8.2.0/lib"
   else
     export
   LD_LIBRARY_PATH="/opt/gcc/8.2.0/lib64/:/opt/gcc/8.2.0/lib:${!LD_LIB_P_VAR}"
   fi
```
## Expected Behavior ##
The `${!LD_LIB_P_VAR}` should read `$LD_LIBRARY_PATH`.

On my machine (a linux VM) `${!LD_LIB_P_VAR}` does not resolve to anything, which means the if test takes the first branch and overwrites the previous contents of `$LD_LIBRARY_PATH`. I suspect you need `${!LD_LIB_P_VAR}` and its associated logic higher up the chain, but that variable should be resolved to the appropriate value before its written to the _generated_ setup.sh file.

While you're at it, you might as well also delete the trailing `/` after `lib64`, so that, for example, the penultimate line reads
```
  LD_LIBRARY_PATH="/opt/gcc/8.2.0/lib64:/opt/gcc/8.2.0/lib:$LD_LIBRARY_PATH"
```

[links]:#
[Google]: https://www.google.com
[Stack Overflow]: https://stackoverflow.com
[fenced codeblocks]: https://help.github.com/articles/creating-and-highlighting-code-blocks/
[GFortran bug tracker]: https://gcc.gnu.org/bugzilla/
[Intel MPI]: https://software.intel.com/en-us/intel-mpi-library
[MPT]: http://www.sgi.com/products/software/sps.html
[MVAPICH]: http://mvapich.cse.ohio-state.edu
[MPICH]: https://www.mpich.org
[Open-MPI]: https://www.open-mpi.org
[CONTRIBUTING.md]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/CONTRIBUTING.md
[Helping Out]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/CONTRIBUTING.md#helping-out
[official release]: https://github.com/sourceryinstitute/OpenCoarrays/releases
[CMake install]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/INSTALL.md#cmake-scripts
[homebrew]: https://brew.sh
[issues page]: https://github.com/sourceryinstitute/OpenCoarrays/issues
[mailing list]: https://groups.google.com/forum/#!forum/opencoarrays
[install.sh]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/install.sh

### PR DESCRIPTION
**Pull request by rouson**
[links]:#
[contributing guidelines]: https://github.com/sourceryinstitute/OpenCoarrays/blob/master/CONTRIBUTING.md
[issue]: https://github.com/sourceryinstitute/OpenCoarrays/issues
[PR response img]: https://img.shields.io/issuestats/p/github/sourceryinstitute/OpenCoarrays.svg?style=flat-square
[coverage]: https://img.shields.io/codecov/c/github/sourceryinstitute/OpenCoarrays/master.svg?style=flat-square

|  Avg response time                |  coverage on master         |
|:---------------------------------:|:---------------------------:|
| ![Issue Stats][PR response img]   | ![Codecov branch][coverage] |

## Summary of changes ##

Deleted '!' in environment variable evaluations in `report_results.sh` installation script.

## Rationale for changes ##

The previous fix didn't work for me on Lubuntu.  

## Additional info and certifications ##

This pull request (PR) is a:

- [X] Bug fix
- [ ] Feature addition
- [ ] Other, Please describe:

### I certify that ###

- [X] I reviewed and followed the [contributing guidelines], including
      - Increasing test coverage for all feature-addition PRs
      - Increasing test coverage for all bug-fix PRs for which there
        does not already exist a related test that failed before the PR
      - At least maintaining test coverage for all other PRs
      - Ensuring that all tests pass when run locally 
      - Naming PR  to indicate work in progress (WIP) and to attach the PR
        to the appropriate bug report or feature request [issue]
      - White space (no trailing white space or white space errors may
        be introduced)
      - Commenting code where it is non-obvious and non-trivial
      - Logically atomic, self consistent and coherent commits
      - Commit message content
      - Waiting 24 hours before self-approving the PR to give another
        OpenCoarrays developer a chance to review my proposed code

### CODE REVIEW

File: .VERSION [hunk 1]
```diff
@@ -1,6 +1,6 @@
 $Format:%d%n%n$
 # Fall back version, probably last release:
-2.3.1
+2.5.0
```
Comment (zbeekman) [inline .VERSION]: This is a duplicate change with your other PR. If you were to have based this branch from master I can do an octopus merge later and merge both at once.

File: prerequisites/install-functions/report_results.sh [hunk 2]
```diff
@@ -69,10 +69,10 @@ report_results()
     if [[ -d "${compiler_install_root%/}/lib" || -d "${compiler_install_root%/}/lib64" ]]; then
       echo "# Prepend the compiler library paths to the ${LD_LIB_P_VAR} environment variable:" | tee -a setup.sh setup.csh
       compiler_lib_paths="${compiler_install_root%/}/lib64/:${compiler_install_root%/}/lib"
-      echo "if [[ -z \"\${!${LD_LIB_P_VAR}}\" ]]; then                                    " >> setup.sh
+      echo "if [[ -z \"\${${LD_LIB_P_VAR}}\" ]]; then                                    " >> setup.sh
```
Comment (zbeekman) [inline prerequisites/install-functions/report_results.sh]: gah! Good catch. You are in fact correct that this is the correct logic to resolve this.

File: .VERSION (hunk 1 above)
Comment (rouson) [reply .VERSION]: Oops.  I just merged the other one and deleted the branch.  I assume there's no problem with merging this one too.  I would expect `git` is smart enough to recognize the duplicate commit and only add in what's new.  I hope it's ok if I go for it.

**Comment by rouson** [comment]
I clicked "Update branch" and will merge this once the tests pass again.

**Pull request merged**